lib/test/unit_test/DomainParticipant/DomainParticipant.cpp

```diff
@@ -0,0 +1,937 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdio>
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+#include <domain_participant/builtin/InitialPeers.hpp>
+#include <domain_participant/builtin/MetatrafficExternalUnicastLocators.hpp>
+#include <domain_participant/DefaultExternalUnicastLocators.hpp>
+#include <domain_participant/DomainParticipant.hpp>
+#include <exception/Exception.hpp>
+
+using namespace eprosima::qosprof;
+using namespace eprosima::qosprof::domain_participant;
+
+class DomainParticipantTests : public ::testing::Test
+{
+protected:
+
+    // Test fixture variables
+    std::string xml_filename_;
+    std::string participant_profile_;
+    std::string another_participant_profile_;
+    std::string participant_name_;
+    std::vector<std::string> valid_values_;
+    std::vector<std::string> invalid_values_;
+    std::vector<std::string> invalid_messages_;
+    std::vector<std::string> print_results_;
+
+    // Functors
+    std::function<std::string(const std::string&, const std::string&, int32_t)> print_functor_;
+    std::function<void(const std::string&, const std::string&, const std::string&)> push_functor_;
+    std::function<void(const std::string&, const std::string&, const std::string&, int32_t)> update_functor_;
+    std::function<void(const std::string&, const std::string&, int32_t)> clear_functor_;
+
+    void SetUp() override
+    {
+        xml_filename_ = "test.xml";
+        participant_profile_ = "test_profile";
+        another_participant_profile_ = "second_test_profile";
+        participant_name_ = "participant_test";
+        valid_values_.clear();
+        invalid_values_.clear();
+        invalid_messages_.clear();
+        print_results_.clear();
+        print_functor_ = nullptr;
+        push_functor_ = nullptr;
+        update_functor_ = nullptr;
+        clear_functor_ = nullptr;
+    }
+
+    void TearDown() override
+    {
+        // Clean test workspace
+        std::remove(xml_filename_.c_str());
+    }
+
+    void print_push_update_clear_test()
+    {
+        // Functors must be initialized
+        ASSERT_NE(print_functor_, nullptr);
```

Comment:
Should these calls be `ASSERT_EQ`? In that case, they would fail if they are not initialized

Reply:
That is exactly what the current command is checking, asserting that the corresponding functor is NOT EQUAL to `nullptr`. The test will fail if the functor is not initialized (is `nullptr`) and it will not keep executing further instructions that will cause a core dumped accessing a non-initialized functor.